api/src/pcapi/scripts/clean_eans.py

```diff
@@ -0,0 +1,151 @@
+- Non LIVRE_PAPIER with 10 numbers ean
+- 13-chars eans with 10 numbers
+
+WARNING : I (vencespass) personally think it is quite important
+to have a look at the existing data before using this algorithm : 
+I belive there are some easily fixable cases that could become lost after
+this script is run.
+Useful commands to check the data :
+select * from product where "jsonData"->>'ean' is not null and (length("jsonData"->>'ean')>13);
+select * from product where "jsonData"->>'ean' is not null and (length("jsonData"->>'ean')<13);
+select * from product where "jsonData"->>'ean' is not null and length("jsonData"->>'ean')=13 and "jsonData"->>'ean' !~ '^[0-9]+$';
+"""
+from itertools import islice
+import re
+from typing import Generator
+
+import sqlalchemy as sa
+
+from pcapi.core.categories.subcategories_v2 import LIVRE_PAPIER
+from pcapi.core.offers import models as offers_models
+from pcapi.models import db
+
+
+def print_and_write(file: str | None, message: str) -> None:
+    print(message, file=open(file, "a", encoding="utf-8"))
+
+
+def clean_product_or_offer(ean_object: offers_models.Product | offers_models.Offer, file: str | None = None) -> None:
+    # Different types of ean to clean :
+    # empty eans ({"ean":""})
+    # Badly formatted eans ({"ean":"9-876543-2198-76"})
+    # old eans, with length of 10 ({"ean":"2394739534"}) WARNING : sometimes, they have "-" too
+    #   - Should add "978" at the beginning if subcategory is "Livre papier"
+
+    # NOT HANDLED HERE:
+    # i should use capslock eans ({"ean":"àéè-_'&éé"})
+    # iDon'tKnowHowToUseAComputer ({"ean":"Author Name", "author"="9876543219876"})
+
+    # --- --- ---
+
+    # Log of what is done :
+    #
+    # NO_EAN, product_id (should not happen)
+    #
+    # Stuff to double check before committing:
+    #   BAD_EAN, product_id, ean
+    #   UPDATED_EAN, product_id, old_ean -> new_ean
+    #   UPDATED_EAN_BOOK, product_id, old_ean -> new_ean
+    #
+    # Stuff for later:
+    #   NOT_UPDATED_EAN_BOOK, product_id, subcategory, old_ean  -> new_ean (what we will need look at later)
+
+    ean = ean_object.extraData.get("ean", None)
+    if ean is None:
+        print_and_write(file, f"NO_EAN, {ean_object.id}")
+        return
+    ean_numbers = [int(s) for s in re.findall(r"\d", ean)]
+    new_ean = "".join([str(number) for number in ean_numbers])
+
+    if len(new_ean) == 13:
+        ean_object.extraData["ean"] = new_ean
+        print_and_write(file, f"UPDATED_EAN, {ean_object.id}, {ean} -> {new_ean}")
+    elif len(new_ean) == 10:
+        if ean_object.subcategoryId == LIVRE_PAPIER.id:
+            new_ean = "978" + new_ean
+            ean_object.extraData["ean"] = new_ean
+            print_and_write(file, f"UPDATED_EAN_BOOK, {ean_object.id}, {ean} -> {new_ean}")
+        else:
+            print_and_write(file, f"NOT_UPDATED_EAN_BOOK {ean_object.id}, {ean_object.subcategoryId}, {ean}")
+    else:
+        if ean == "":
+            # We know what to do, no need to log
```

Comment:
```suggestion
            # We don't know what to do, no need to log
```
Ou alors j'ai pas bien compris.
Mais si l'ean est vide, ne veut on pas plutôt le passer à None ?

Reply:
Ah yes, on le drop 3 lignes en dessous, je vire le return, my bad :roll_eyes: